### PR TITLE
Fixes #7041: Adds an attribute to change login prompt editText cursor color

### LINKED ISSUE #7041: Cursor not visible in TextInputEditText from login prompt
The cursor is not visible at all on light/dark themes.

For more informations see [Fenix issue](https://github.com/mozilla-mobile/fenix/issues/10113).

### PR DESCRIPTION
**Pull request by ValentinTimisica**
---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks.
- [x] **Tests**: This PR does not include thorough tests. Small UI change.
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md).
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md).

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/prompts/src/main/res/layout/mozac_feature_prompt_login_prompt.xml [hunk 1]
```diff
@@ -86,7 +86,6 @@
 
         <com.google.android.material.textfield.TextInputEditText
             android:id="@+id/password_field"
-            style="@style/Widget.MaterialComponents.TextInputLayout.OutlinedBox"
```
Comment (Amejia481) [inline components/feature/prompts/src/main/res/layout/mozac_feature_prompt_login_prompt.xml:89]: Should we use `MozacPromptLoginTextInputLayoutCursorAppearance` style here? I don't see how they linked :)

Comment (Amejia481) [reply components/feature/prompts/src/main/res/layout/mozac_feature_prompt_login_prompt.xml:89]: Oh I see it gets its value from the parent that has the style `MozTextInputLayout`, right?

Comment (ValentinTimisica) [reply components/feature/prompts/src/main/res/layout/mozac_feature_prompt_login_prompt.xml:89]: Right, from the parent.

File: components/feature/prompts/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
+        <item name="android:theme">@style/MozacPromptLoginTextInputLayoutCursorAppearance</item>
     </style>
     <style name="MozDialogStyle" parent="Theme.Design.Light.BottomSheetDialog">
         <item name="android:windowIsFloating">false</item>
         <item name="android:windowSoftInputMode">adjustResize</item>
     </style>
-</resources>
\ No newline at end of file
+    <style name="MozacPromptLoginTextInputLayoutCursorAppearance" parent="ThemeOverlay.MaterialComponents.TextInputEditText.OutlinedBox">
+        <item name="colorControlActivated">?mozacPromptLoginEditTextCursorColor</item>
```
Comment (Amejia481) [inline components/feature/prompts/src/main/res/values/styles.xml:16]: Where are we proving the value for `?mozacPromptLoginEditTextCursorColor`? As we don't have a default value what will happen for consumer that doesn't provide this value?

Comment (Amejia481) [reply components/feature/prompts/src/main/res/values/styles.xml:16]: Do you have a sample code where we are providing the value

Comment (ValentinTimisica) [reply components/feature/prompts/src/main/res/values/styles.xml:16]: I use the attribute in Fenix in styles.xml like this:
```<item name="mozacPromptLoginEditTextCursorColor">@color/prompt_login_edit_text_cursor_color_normal_theme</item>```
The same for private mode.

Regarding the default value, I tested on Fenix and if I don't use at all 
`?mozacPromptLoginEditTextCursorColor` it works as before (the cursor color is the same as before - the color from the [Fenix issue](https://github.com/mozilla-mobile/fenix/issues/10113)). But I don't know what will happen for other apps.

Any suggestions? I'm open to any change.

Comment (Amejia481) [reply components/feature/prompts/src/main/res/values/styles.xml:16]: No problem, I just wanted to confirm that we don't have a crash when we don't provide a value. 👍

**Pull request merged**